consensus/ibftlegacy/src/main/java/tech/pegasys/pantheon/consensus/ibftlegacy/headervalidationrules/VoteValidationRule.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2018 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.consensus.ibftlegacy.headervalidationrules;
+
+import tech.pegasys.pantheon.consensus.ibftlegacy.IbftLegacyVotingBlockInterface;
+import tech.pegasys.pantheon.ethereum.core.BlockHeader;
+import tech.pegasys.pantheon.ethereum.mainnet.DetachedBlockHeaderValidationRule;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class VoteValidationRule implements DetachedBlockHeaderValidationRule {
```

Comment:
If you passed in an instance of `VotingBlockInterface` this looks like it would work with both IBFT and Clique. Might be a follow-up PR mind you.

Reply:
Mind if I let this one go for now (but it would be seriously nice).